### PR TITLE
SAN Amendments

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#15189

*Description of changes:*
- Implements Amendment framework/submissions
- Notice and Order property tracking and badges in existing SAN Panels

Screenshot 2024-06-04 at 1 51 12 PM

Screenshot 2024-06-04 at 1 50 37 PM

- Amendments Review variations

Screenshot 2024-06-04 at 1 51 33 PM

- Table alignment fixes for Ppr parties

Screenshot 2024-06-04 at 1 47 05 PM

- Registration Table Mapping

Screenshot 2024-06-04 at 1 49 28 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1913-xls4tlfe.web.app

File: ppr-ui/src/utils/registration-helper.ts [hunk 1]
```diff
@@ -51,6 +52,43 @@ export function setAmendmentList (baseList:Array<any>, addList:Array<any>, delet
...
+ *
+ * @returns An object containing the updated `addNotices` and `deleteNotices` arrays.
+**/
+ export function setNoticeAmendments (
+  notices: Array<AddEditSaNoticeIF>, addNotices: Array<AddEditSaNoticeIF>, deleteNotices: Array<AddEditSaNoticeIF>
+): { addNotices: Array<AddEditSaNoticeIF>; deleteNotices: Array<AddEditSaNoticeIF> } {
+  notices?.forEach(notice => {
+    // Include Added/Amended Notices in ADD block
+    if ([ActionTypes.ADDED, ActionTypes.EDITED].includes(notice?.action)) {
+      console.log(notice)
```
Comment (dimak1) [inline ppr-ui/src/utils/registration-helper.ts]: Not sure if left here by mistake.

File: ppr-ui/src/views/amendment/AmendRegistration.vue [hunk 2]
```diff
@@ -503,6 +512,12 @@ export default defineComponent({
 
     const hasAmendmentChanged = (): boolean => {
       let hasChanged = false
+
+      if (isSecurityActNotice.value) {
+        if (!isEqual(getSecuritiesActNotices.value, getOriginalSecuritiesActNotices.value)) {
```
Comment (dimak1) [inline ppr-ui/src/views/amendment/AmendRegistration.vue]: These two conditions could be combined into one.

**Review by doug-lovett — APPROVED** [review]
Looks great.

**Review by dimak1 — APPROVED** [review]
Lots of work. Besides few small things, looks good!

File: ppr-ui/src/utils/registration-helper.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/utils/registration-helper.ts]: Definitely, thank you!

File: ppr-ui/src/views/amendment/AmendRegistration.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/amendment/AmendRegistration.vue]: Definitely, will do, thank you!

**Review by dimak1 — APPROVED** [review]

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-assets-dev--pr-1913-xls4tlfe.web.app

**Pull request merged**